### ISSUE TITLE
pub get is not working saying (This is taking an unexpectedly long time.) 

### ISSUE THREAD

**Issue by DanGerHOGGISH** [flutter/flutter#63174]
Good afternoon.
I'm having an issue with pub get. I tried many things(reinstalled the Android Studio even i installed Visual Studio and it did not work anyway then i begin to think it might be the problem with the flutter SDK. I also reinstalled that but it did not work I alsa reinstalled the Dart SDK and gave the path to Android Studio. File->Settings->Dart(Flutter) but it did not work.
here is the errors and pub get is not running.
![error](https://user-images.githubusercontent.com/67454411/89620312-ebc6c880-d8a8-11ea-897f-a86dc3043996.jpg)

**Comment by amanv8060** [comment]
can you please provide the output of `flutter doctor`

**Comment by DanGerHOGGISH** [comment]
> can you please provide the output of `flutter doctor`
yes sure thanks for your replay
![cmd](https://user-images.githubusercontent.com/67454411/89620420-0b5df100-d8a9-11ea-9086-ee63702a6f06.jpg)

**Comment by DanGerHOGGISH** [comment]
name: food_delivery
description: Food Delivery

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.3
  carousel_pro: ^1.0.0

dev_dependencies:
  flutter_test:
    sdk: flutter

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  assets:
    - assets/images/
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:

  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
i could not find the file and copied it

**Comment by DanGerHOGGISH** [comment]
[pubspec.zip](https://github.com/flutter/flutter/files/5039924/pubspec.zip)

**Comment by amanv8060** [comment]
There looks no issue , this happens  sometime when the internet issue is there  try `ping pub.dartlang.org `  to see if your internet connection is there , you could also try `pub get --trace` to see the work of pub get , post the output of both of them here

**Comment by DanGerHOGGISH** [comment]
> There looks no issue , this happens sometime when the internet issue is there try `ping pub.dartlang.org ` to see if your internet connection is there , you could also try `pub get --trace` to see the work of pub get , post the output of both of them here
![trace](https://user-images.githubusercontent.com/67454411/89622309-4ca3d000-d8ac-11ea-94a5-bf82dc5b556c.jpg)
![trace1](https://user-images.githubusercontent.com/67454411/89622315-4dd4fd00-d8ac-11ea-8838-eba66bb4494b.jpg)

**Comment by DanGerHOGGISH** [comment]
![waiting](https://user-images.githubusercontent.com/67454411/89622441-807ef580-d8ac-11ea-9488-7a832c7a7adc.jpg)
I am waiting like 20-30minutes it does not create a flutter project online if i tick tick the offline project it creates what is the problem

**Comment by amanv8060** [comment]
Issue #25433 that has mention of the problem that arises when you are behind proxy . Do have a look

**Comment by pedromassangocode** [comment]
Hi @DanGerHOGGISH 
Can you please run (making sure you run from your terminal and not from an IDE)
```console
cd <your app folder>
flutter doctor -v
flutter pub upgrade --verbose
``` 
and provide the output.
Thank you

**Comment by DanGerHOGGISH** [comment]
> ```
> flutter pub upgrade --verbose
> ```

Thank you for your reply! I dont know how to provide the output I couldn't run the pub get or pub upgrade then i tried to upgrade Android Studio && FLutter SDK $$ Dart SDK and now when i try to create a Flutter project it just shows loading pop up https://user-images.githubusercontent.com/67454411/89622441-807ef580-d8ac-11ea-9488-7a832c7a7adc.jpg
and this one i  opened the github project that i downloaded from github and it shows this error. Actually i cannot create a ne project https://user-images.githubusercontent.com/67454411/89620312-ebc6c880-d8a8-11ea-897f-a86dc3043996.jpg

**Comment by pedromassangocode** [comment]
Hi @DanGerHOGGISH 
This maybe caused by internet connection issue, can you try a different network and try again?
Thank you

**Comment by DanGerHOGGISH** [comment]
> Hi @DanGerHOGGISH
> This maybe caused by internet connection issue, can you try a different network and try again?
> Thank you

but still this problem i had been using the exact same network. why this problem occured ? i dont understand i used VPN but could not create a project. 2-3 weeks ago when i typed pub get it did not get that package and i reinstalled everything flutter sdk Android Studio even downloaded the VS code but could not even create a new project i dont think it is network problem

**Comment by DanGerHOGGISH** [comment]
Good morning I found the issue the version of the Flutter SDK 1.20.0 is kind of buggy that is why it did not work when the new version was released(1.20.1) it worked fine thanks /close
